Fix singleton (#4)
design-patterns-for-humans/vala: README.md

Singleton pattern is actually considered an anti-pattern and overuse of it should be avoided. It is not necessarily bad and could have some valid use-cases but should be used with caution because it introduces a global state in your application and change to it in one place could affect in the other areas and it could become pretty difficult to debug. The other bad thing about them is it makes your code tightly coupled plus it mocking the singleton could be difficult.

> **Progromatic Example**

Reviewer: Hi, tnx for contrib. Pse fix Programmatic

To create a singleton there are two main ways to do it when targeting GLib (the stanard target),

> thu first is to use the `[SingleInstance]` attributes:

Reviewer: "the first..."

**Programmatic Example**

> To create a singleton there are two main ways to do it when targeting GLib (the stanard target),
> the first is to use the `[SingleInstance]` attributes:

Reviewer 2: ```suggestion
When targeting GLib (the standard target), there are two ways of creating a
singleton. The first one is using the `[SingleInstance]` attribute:
```

Reviewer: Thanks,! Added your contribution manually. BR

```vala
[SingleInstance]
public class President : Object  {
    // No default clone and unserialize methods.
}
```

> The other way is to use `GLib.Once`:

Reviewer 2: ```suggestion
The second one is to use `GLib.Once`:
```

Reviewer: Ditto :)
